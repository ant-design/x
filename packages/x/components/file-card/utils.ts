export function getSize(size: number) {
  let retSize = size;
  const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB'];
  let unitIndex = 0;

  while (retSize >= 1024 && unitIndex < units.length - 1) {
    retSize /= 1024;
    unitIndex++;
  }

  return `${retSize.toFixed(0)} ${units[unitIndex]}`;
}

export function matchExt(suffix: string, ext: string[]) {
  return ext.some((e) => suffix.toLowerCase() === `.${e}`);
}
