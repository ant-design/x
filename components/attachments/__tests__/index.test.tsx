import React from 'react';

import Attachments from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import { fireEvent, render } from '../../../tests/utils';

describe('attachments', () => {
  mountTest(() => <Attachments />);
  rtlTest(() => <Attachments />);
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });
});
