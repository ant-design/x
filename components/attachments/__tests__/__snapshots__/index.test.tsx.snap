// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attachments rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-attachment ant-attachment-rtl"
  dir="rtl"
>
  <div
    class="ant-attachment-placeholder"
  >
    <span
      class="ant-upload-wrapper ant-upload-rtl"
    >
      <div
        class="ant-upload ant-upload-drag ant-upload-rtl"
        style="padding: 0px; border: 0px; background: transparent;"
      >
        <span
          class="ant-upload ant-upload-btn"
          role="button"
          tabindex="0"
        >
          <input
            accept=""
            name="file"
            style="display: none;"
            type="file"
          />
          <div
            class="ant-upload-drag-container"
          >
            <div
              class="ant-attachment-placeholder-inner ant-flex ant-flex-align-center ant-flex-justify-center ant-flex-rtl ant-flex-vertical"
            >
              <span
                class="ant-typography ant-typography-rtl ant-attachment-placeholder-icon"
              />
              <h5
                class="ant-typography ant-typography-rtl ant-attachment-placeholder-title"
              />
              <span
                class="ant-typography ant-typography-rtl ant-typography-secondary ant-attachment-placeholder-description"
              />
            </div>
          </div>
        </span>
      </div>
    </span>
  </div>
</div>
`;
